netshare/learn/utils/word2vec_embedding.py

```diff
@@ -35,4 +46,99 @@ def word2vec_train(
         model.save(model_path)
     logger.info(f"Word2Vec model is saved at {model_path}")
 
-    return model_path
+    return None
+
+
+def get_word2vec_type_col(word2vec_cols: List[Any]) -> Dict[str, List[str]]:
+    dict_type_cols: Dict[str, List[str]] = {}
+    for col in word2vec_cols:
+        type = col.encoding
+        if type not in dict_type_cols:
+            dict_type_cols[type] = []
+        dict_type_cols[type].append(col.column)
+    return dict_type_cols
+
+
+def build_annoy_dictionary_word2vec(
+    df: pd.DataFrame,
+    model_path: str,
+    word2vec_cols: List[Any],
+    word2vec_size: int,
+    n_trees: int,
+) -> None:
+
+    dict_encoding_type_vs_idx_ele_dict_pair = {}
+
+    model = Word2Vec.load(model_path)
+    wv = model.wv
+
+    # encoding type and column names may not be necessarily the same. eg. for a pcap/netflow dataset
+    # encoding type = word2vec_port -> column: ["srcport", "dstport"]
```

Comment:
so the type is structured from two different entities? 
`word2vec` which tells us the type, and `port`/`proto` which tells us the "group of words" that it should be included in?
Why isn't that another key in the field configuration, and just loaf it as a member of `Word2VecField`?

Reply:
And then, you can use `word2vec_fields: List[Field]` instead of `word2vec_cols: List[Any]` everywhere.

Reply:
But: Maybe this is not in the scope of this PR, and we should do this change later.
It's your choice :)

Reply:
Great idea! i think so. Let me mark it and put it into another PR